job-server/src/main/scala/spark/jobserver/JobManagerActor.scala

```diff
@@ -115,10 +124,15 @@ class JobManagerActor(daoActor: ActorRef, supervisorActorAddress: String, contex
   private var contextConfig: Config = _
   private var contextName: String = _
   private var isAdHoc: Boolean = _
-  private var statusActor: ActorRef = _
+  @VisibleForTesting
```

Comment:
What is this VisibleForTesting?

Reply:
Visibility of some of the variables were changed from private to protected so that we can have access to them in testcases. Normally, it is not too obvious. `@VisibleForTesting` makes it obvious that we changed the visibility just for this purpose. This is just for documentation purpose.